cue.c

```diff
@@ -479,9 +479,26 @@ struct cue_sheet *cue_from_file(const char *file)
 {
 	ssize_t size;
 	char *buf = mmap_file(file, &size);
+
 	if (size == -1)
 		return NULL;
+
+	bool bom_detected = false;
+
+	// ignore BOM
+	if (size >= 3 && memcmp(buf, "\xEF\xBB\xBF", 3) == 0) {
+		buf += 3;
+		size -= 3;
+		bom_detected = true;
+	}
+
 	struct cue_sheet *rv = cue_parse(buf, size);
+
+	if (bom_detected) {
+		buf -= 3;
+		size += 3;
+	}
```

Comment:
This is a bit ugly, to be honest. Better to create new variables specifically for `cue_parse()`, and avoid mandatory fragile "cleanup" code.